### PR TITLE
refactor: Move node and wallet code out of src/interfaces

### PR DESCRIPTION
**Pull request by ryanofsky**
This PR is part of the [process separation project](https://github.com/bitcoin/bitcoin/projects/10).

---

Move `NodeImpl` from `interfaces/node.cpp` to `node/interfaces.cpp`
Move `ChainImpl` from `interfaces/chain.cpp` to `node/interfaces.cpp`
Move `WalletImpl` from `interfaces/wallet.cpp` to `wallet/interfaces.cpp`

No changes to any classes (can review with `git diff --color-moved=dimmed_zebra`)

Motivation for this change is to move node and wallet code to respective directories where it might fit in better than `src/interfaces/`, but also to remove all unnecessary code from `src/interfaces/` to unblock #19160 review, which has been hung up partially because of code organization. Building on top of this PR, #19160 should now be able to organize interface implementations more understandably in `src/node/` `src/wallet/` `src/ipc/` and `src/init/` directories instead of having so much functionality all in `src/interfaces/`

### CODE REVIEW

**Review by promag — COMMENTED** [review]
Code review ACK 629a9299b2a7241a3fa7d597cb34abcbe1af9255.

This change is reasonable, interface implementations are moved to their "home".

File: src/node/interfaces.cpp [hunk 1]
```diff
@@ -2,47 +2,58 @@
 // Distributed under the MIT software license, see the accompanying
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
-#include <interfaces/node.h>
-
 #include <addrdb.h>
 #include <banman.h>
+#include <boost/signals2/signal.hpp>
```
Comment (MarcoFalke) [inline src/node/interfaces.cpp:7]: why is this moved?

**Review by maflcko — APPROVED** [review]
review ACK 629a9299b2a7241a3fa7d597cb34abcbe1af9255 🔺

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

review ACK 629a9299b2a7241a3fa7d597cb34abcbe1af9255 🔺
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUgpKgv9FXDr3pKOUjSzUhQP8v6tY0NTOf/V0UP6izQZuoOkAxjFJq/FgXNAgnU7
NyX7VHnqx/1gf+GAOpKrcxvWderBde6fDPNHS+jcZo4aNsv15Vsqv/Fia9VaKw0h
kBkEzWm8N3hEktVdqTsZrGee26qD0jlJM64eW/Se9Kq3D6uQgyusEOJ5uiu6cmG4
ycXeRnTCs1PCneNlt/6hypLs9iSe5RUHhM8rmEFh6HLBmW0t+mpca7cOdGmZAGXp
R+rMTh8LeljiEL10I/v1pg0OYDOIJ7IQpERwrajiB4eWVgHbHmxqr1AVGikQDm36
StCw6QRCV0GzRPE963zpP5SvI0JxdOTTrTBzlKZqCMO5b1hoZJDq/9NEUIZQWIVq
4nYDDtyrQBqSunNAqEi1QAmEbt+IhGZR4v0yWGZHLLOuAXOAs+X5GbOwRzgPQeAC
rx9EawhmAxgH1tkSUScC3xyYgeN5GGosF9yX8xrocoQNKbILu1E8y1u1tt1Izjkx
SAS0x21D
=i4RY
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `bfed3f19ad97ec6b521680bc88144a3f4a28fae8cda34c721b3183fddf8c449c  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e892940108bfed3f19ad97ec6b521680bc88144a3f4a28fae8cda34c721b3183fddf8c449cf0102ac48ce448ca38346fc7f17e6fdde8e408fff01028e2fc1b990052cdf71b2e3dc16a874008f1045fc60105f00853418cb2bb79800b0083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff010864225a2a5ed50f311d6f13a65087cd908f1045fc60105f00896b840278e1ad5400083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff01098a4144fe5e779086f4783050f8d555c08f1045fc60105f008b1e7805009ebdd650083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6df010de62fc5809325999598016a937f8a72f08f1045fc60108f008768849954bb5cc5b0083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6d)

</details>

File: src/node/interfaces.cpp (hunk 1 above)
Comment (ryanofsky) [reply src/node/interfaces.cpp:7]: re: https://github.com/bitcoin/bitcoin/pull/20494#discussion_r533155930

> why is this moved?

Not sure. This came from https://github.com/bitcoin/bitcoin/pull/19160 where I was using IWYU a lot. I tried to manually sort system includes below project includes after running it, but I wasn't very diligent. It'd be more convenient if there was just a single include section, or an automatic include sorter
